holder.yml

```diff
@@ -100,17 +100,17 @@ paths:
           description: Service not implemented.
         "500":
           description: Internal server error.
-  /exchanges/{exchange-id}/{transaction-uuid}:
+  /exchanges/{exchange-id}/{transaction-id}:
```

Comment:
I still don't understand or plan to use transaction id.

Reply:
> I still don't ... plan to use transaction id.

This is fine.

> I still don't ... understand transaction id.

This is concerning, my hope is that we can at least get you to understanding it so that you use it if it ever becomes useful to you.

Digital Bazaar finds transaction id useful to refer to any sort of stateful binding on the server side. For example, when the client is like: "I want to do an exchange of a specific type with you.", and the server responds with "Ok, here's a VPR containing a challenge", the server *might* need to store that challenge and information specific to that particular transaction server-side. It can store this indexed against a transaction-id (or something derived from the transaction-id... transaction-id + nonce, for example).

Does that make sense?

Reply:
> @msporny so `transaction-id` is like `domain` in the VC Data Model?

Hrm, nope... the client sets domain, not the server. If the server sets domain, it can attack the client by replaying challenge/domain combinations that it's doing behind the scenes w/ other websites. That is, if we let the server set the domain, it's an attack vector for any client doing DID Auth.

Reply:
client server language is problematic here... I think you mean that the `verifier` sets the `domain` and `challenge` for a presentation... and my question was, are either of those terms related to these new identifiers... seems like the answer is not, and you would expect a verifier to receive many presentations with a unique domain and challenge values to the same `transaction-id`.

Reply:
in trace interop profile we use the term `definition` for `{exchange-id}` and `instance` for `{transaction-id}`... I wonder if its too late to change the names here to align with that...

```
{exchange-id} -> {definition-id}
```

```
{transaction-id} -> {instance-id}
```

Reply:
> I think you mean that the verifier sets the domain and challenge for a presentation (and the holder signs over them)

No, I definitely don't mean that. That could become a security vulnerability. verifier app/service sets the challenge, holder app/service sets the domain and signs over both values.

> are either of those terms related to these new identifiers... seems like the answer is no

The answer isn't "no", it's: They might be associated in a 1-to-1 fashion, or they might be associated in an n-to-1 fashion.

> you would expect a verifier to receive many presentation exchanges with a unique domain and challenge values to the same transaction-id

Digital Bazaar doesn't have that use case currently, but we do admit that it's possible that some verifiers might structure their management of transaction-ids in this way. For example, if a single transaction requires N holders to submit VPs to a single transaction-id-identified endpoint.

Reply:
> aud MUST represent (i.e., identify) the intended audience of the [verifiable presentation](https://www.w3.org/TR/vc-data-model/#dfn-verifiable-presentations) (i.e., the [verifier](https://www.w3.org/TR/vc-data-model/#dfn-verifier) intended by the presenting [holder](https://www.w3.org/TR/vc-data-model/#dfn-holders) to receive and verify the [verifiable presentation](https://www.w3.org/TR/vc-data-model/#dfn-verifiable-presentations)).

We should probably update the definition here: https://w3c-ccg.github.io/security-vocab/#domain

If a verifier suggests a domain and the holder signs a presentation over a different value, what happens?

Reply:
> AFAIK, there is no mutual authentication happening here

You could argue that mutual auth /is/ happening *if* the client is paying attention to TLS (which is almost certainly is doing these days)... and the server is ensuring the DID Auth response is associated with a specific DID.

> and domain and challenge don't actually protect against anything other than replay attacks on the verifiers web server. 

If the client sets the `domain`, then the client is protecting itself from replays across the entirety of the Internet (if the domain value maps to DNS) and as long as the server is protecting itself from replays by paying attention to the domain value.

> (assuming that challenges are stateful and consumed, which also implies a DDoS vector)

Yep, it is a DDoS vector if the server doesn't follow normal DDoS-protection hygiene (limit requests by IP, HTTP Header, challenge-under-attack, or any other form of preliminary client identification).